core/node/reorg_detector/src/lib.rs

```diff
@@ -329,7 +329,8 @@ impl ReorgDetector {
                     .update_correct_block(checked_l2_block, checked_l1_batch);
                 None
             } else {
-                let diverged_l1_batch = checked_l1_batch;
+                let diverged_l1_batch =
+                    checked_l1_batch + max(root_hashes_match, commitments_match) as u32;
```

Comment:
why not `+ (root_hashes_match || commitments_match) as u32`

Reply:
it's the same.